src/R/Components/Impl/History/Implementation/RHistoryProvider.cs

```diff
@@ -16,18 +16,29 @@ namespace Microsoft.R.Components.History.Implementation {
     [Export(typeof(IRHistoryProvider))]
     internal class RHistoryProvider : IRHistoryProvider {
         private readonly ITextBufferFactoryService _textBufferFactory;
-        private readonly IFileSystem _fileSystem;
         private readonly IEditorOperationsFactoryService _editorOperationsFactory;
         private readonly IRtfBuilderService _rtfBuilderService;
         private readonly ITextSearchService2 _textSearchService;
         private readonly IRSettings _settings;
         private readonly IContentTypeRegistryService _contentTypeRegistryService;
         private readonly Dictionary<ITextBuffer, IRHistory> _histories;
 
+        private IFileSystem _fileSystem;
+        internal IFileSystem FileSystem {
```

Comment:
They aren't identical. Runtime container provides more exports and may provide them in a different time periods.
Injecting `IFileSystem` in constructor simplifies test code. For example, `RInteractiveWorkflow` calls `CreateRHistory` right in constructor, so we have to explicitly call `_exportProvider.GetExportedValue<IRHistoryProvider>().FileSystem = FileSystemStubFactory.CreateDefault()` in test constructors.
I don't see big difference in default value checking inside constructor or in the internal property. In both cases we know that it is null in production and not null in unit tests.